src/operator/tensor/dot-inl.h

```diff
@@ -775,13 +777,35 @@ inline void DotCsrDnsDnsImpl(const OpContext& ctx,
   }
 
   using nnvm::dim_t;
-
+#if (MSHADOW_USE_MKL == 1)
+  TShape lhs_shape = lhs.shape();
+  TShape rhs_shape = rhs.shape_;
+#endif
   const TBlob data_l = lhs.data();
   const TBlob indptr_l = lhs.aux_data(csr::kIndPtr);
   const TBlob col_idx_l = lhs.aux_data(csr::kIdx);
   const TBlob& data_r = rhs;
   const TBlob data_out = *ret;
 
+#if (MSHADOW_USE_MKL == 1)
+  if (data_l.type_flag_ == mshadow::kFloat32
```

Comment:
I have some questions regarding the csr format in MKL:
- does it allow duplicate entries? 
- does it expect the col indices of any input/output csr matrices to be sorted?

For example, the following csr_matrix is valid in scipy, but not expected in MXNet (https://github.com/apache/incubator-mxnet/blob/master/python/mxnet/ndarray/sparse.py#L295-L296):
```
>>> import scipy.sparse as spsp
>>> a = spsp.csr_matrix(([0.1,0.2,0.4], [3,2,3], [0,3]))
>>> a.toarray()
array([[0. , 0. , 0.2, 0.5]])
```

Is MKL possible to output a csr that will be invalid in MXNet?

Reply:
MKL is the same with mxnet
https://software.intel.com/en-us/mkl-developer-reference-c-sparse-blas-csr-matrix-storage-format
it do not allow duplicate entries & expect the col indices of any input/output csr matrices to be sorted， Element i of the integer array columns is the number of the column in A that contains the i-th value in the values array.